src/Microsoft.AspNet.Mvc.Core/Controllers/DefaultControllerActionArgumentBinder.cs

```diff
@@ -229,18 +221,16 @@ private void ActivateProperties(object controller, Type containerType, Dictionar
             }
         }
 
-        private OperationBindingContext GetOperationBindingContext(
-            ActionContext actionContext,
-            ActionBindingContext bindingContext)
+        private OperationBindingContext GetOperationBindingContext(ControllerContext context)
         {
             return new OperationBindingContext
             {
-                InputFormatters = bindingContext.InputFormatters,
-                ModelBinder = bindingContext.ModelBinder,
-                ValidatorProvider = bindingContext.ValidatorProvider,
+                InputFormatters = context.InputFormatters,
+                ModelBinder = new CompositeModelBinder(context.ModelBinders),
+                ValidatorProvider = new CompositeModelValidatorProvider(context.ValidatorProviders),
                 MetadataProvider = _modelMetadataProvider,
-                HttpContext = actionContext.HttpContext,
-                ValueProvider = bindingContext.ValueProvider,
+                HttpContext = context.HttpContext,
+                ValueProvider = new CompositeValueProvider(context.ValueProviders),
```

Comment:
The composite needs to be created at some point. I chose that boundary to be the static methods on `ModelBindingHelper` and the creation of `OperationBindingContext`. 

If anyone wants to make another suggestion, make your case :laughing: